test/sanity_check.sh

```diff
@@ -5,23 +5,42 @@
 set -e
 
 whitely echo Ping each host from the other
+
 for host in $HOSTS; do
     for other in $HOSTS; do
-        [ $host = $other ] || run_on $host $PING $other
+        if [ "$host" != "$other" ]; then
+            echo $(run_on $host $PING $other) &
+            pids="$pids $!"
+        fi
     done
 done
+for pid in $pids; do wait $pid; done
+unset pids
+
 
 whitely echo Check we can reach docker
 
+function check_docker() {
+    docker_version=$(docker_on $1 version)
+    docker_info=$(docker_on $1 info)
+    docker_weave_version=$(docker inspect -f {{.Created}} weaveworks/weave:${WEAVE_VERSION:-latest})
+    weave_version=$(weave_on $1 version)
+    cat << EOF
+
+Host Version Info: $1
+=====================================
+# docker version
+$docker_version
+# docker info
+$docker_info
+# weave version
+$docker_weave_version
+$weave_version
+EOF
+}
+
 for host in $HOSTS; do
-    echo
-    echo Host Version Info: $host
-    echo =====================================
-    echo "# docker version"
-    docker_on $host version
-    echo "# docker info"
-    docker_on $host info
-    echo "# weave version"
-    docker inspect -f {{.Created}} weaveworks/weave:${WEAVE_VERSION:-latest}
-    weave_on $host version
+    check_docker $host &
```

Comment:
I think running all these in parallel risks the output being interleaved and hard to decode.
Could put the hostname in front of each line.
`docker version` has a `-v` option to control the output, hence put more on each line; `docker info` does not.
I wonder how much value we get out of `docker info` here